### PR TITLE
Verrazzano API Proxy SA validation

### PR DESCRIPTION
**Pull request by bhatpmk**
# Description

This change validates that the roles bound to ServiceAccount for Verrazzano API proxy. The validation is done in a shell script and is included in the Jenkins pipeline including role based tests, for now.

Fixes VZ-2462

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: tools/scripts/api-proxy-validation.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  fi
+  #echo "Secret from the API proxy pod ${secret_name_from_pod}"
+
+  if [ "${sa_secret}" != "${secret_name_from_pod}" ]; then
+    echo "The secret name of ServiceAccount $VERRAZZANO_API differs from the secret obtained from the pod"
+    exit 1
+  fi
+
+  #Get ClusterRoleBindings for verrazzano-api and describe
+  local cluster_role_name_sa="$(kubectl get clusterrolebindings -n $VERRAZZANO_NS -o custom-columns="NAME:metadata.name,SERVICE_ACCOUNTS:subjects[?(@.kind==\"$SERVICE_ACCOUNT\")].name" | grep "${VERRAZZANO_API}" | awk {'print $1'})"
```
Comment (wmhopkins) [inline tools/scripts/api-proxy-validation.sh:59]: Need to support the case where there are multiple bindings for the service account.

File: tools/scripts/api-proxy-validation.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    if [[ -z "${cluster_role_name_sa}" ]]; then
+    echo "Failure to get the role binding for the Service Account."
+    exit 1
+  fi
+  cluster_role_name_sa="${cluster_role_name_sa}"
+  cluster_role_name_using_subjects=$(get_roleRef_name_for_subjects)
+    if [ "${cluster_role_name_sa}" != "${cluster_role_name_using_subjects}" ]; then
+    echo "Failed to match the cluster role using based on the subjects Kind:ServiceAccount for $cluster_role_name_sa"
```
Comment (wmhopkins) [inline tools/scripts/api-proxy-validation.sh:67]: Might be helpful to capitalize FAIL for this case, and add a message with PASS or SUCCEED for the case that only the expected role is bound.

File: tools/scripts/api-proxy-validation.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    exit 1
+  fi
+}
+
+validate_api_proxy_service_account
```
Comment (wmhopkins) [inline tools/scripts/api-proxy-validation.sh:72]: For extra credit, you could verify that the impersonate role contains only the expected impersonate permissions.

**Review by wmhopkins — COMMENTED** [review]
Overall, this looks great; did have a couple comments.

File: tools/scripts/api-proxy-validation.sh (hunk 1 above)
Comment (bhatpmk) [reply tools/scripts/api-proxy-validation.sh]: I have made the necessary changes to handle multiple bindings

File: tools/scripts/api-proxy-validation.sh (hunk 2 above)
Comment (bhatpmk) [reply tools/scripts/api-proxy-validation.sh]: Done.

File: tools/scripts/api-proxy-validation.sh (hunk 3 above)
Comment (bhatpmk) [reply tools/scripts/api-proxy-validation.sh]: Done

**Review by wmhopkins — APPROVED** [review]
Looks good to me.

**Pull request merged**